### ISSUE TITLE
Getting a db error when editing a message

### ISSUE THREAD

**Issue by berkerY** [meeting-room-booking-system/mrbs-code#3680]
I get the following db error when editing a message:

~~~text
Uncaught exception 'MRBS\DBException' in /srv/www/nginx_https/htdocs/mrbs_ulb_dev/web/lib/MRBS/DB.php at line 145
SQLSTATE[23505]: Unique violation: 7 ERROR:  duplicate key value violates unique constraint "mrbs_uq_variable_name"
DETAIL:  Key (variable_name)=(message) already exists.
SQL: INSERT INTO "public"."mrbs_variables"                        ("variable_name", "variable_content")                 VALUES (:p0, :p1) ON CONFLICT ("id") DO UPDATE SET "variable_name"=:p0, "variable_content"=:p1 RETURNING id
Params: Array
(
    [:p0] => message
    [:p1] => {"text":"Das ist eine Message f\u00fcr einen Raum.","from":"2024-06-05T00:00:00","until":"2024-06-07T00:00:00"}
)

#0 /srv/www/nginx_https/htdocs/mrbs_ulb_dev/web/lib/MRBS/Message.php(76): MRBS\DB->command('INSERT INTO "pu...', Array)
#1 /srv/www/nginx_https/htdocs/mrbs_ulb_dev/web/edit_message_handler.php(31): MRBS\Message->save()
#2 {main}
~~~

**Comment by berkerY** [comment]
it's already solved :) You can close the issue.
... you should change the conflict parameter for upsert function in Message.php as follow:

public function save() : bool
  {
    ....
    $sql = db()->syntax_upsert($data, _tbl('variables'), $sql_params, **['variable_name']**, ['id'], true);
    ....
  }

**Comment by berkerY** [comment]
closed

**Comment by berkerY** [comment]
You may want to fix the bug first for your code and close this.

**Comment by campbell-m** [comment]
Thanks.  Now fixed in [46bb4d4](https://github.com/meeting-room-booking-system/mrbs-code/commit/46bb4d440da1a5e66550d5fb89bc40c9222e80a0).
